Formula/agda.rb

```diff
@@ -42,10 +31,12 @@ class Agda < Formula
     end
   end
 
+  deprecated_option "without-malonzo" => "without-ghc"
+
   option "without-stdlib", "Don't install the Agda standard library"
-  option "without-malonzo", "Disable the MAlonzo backend"
+  option "without-ghc", "Disable the GHC backend"
 
-  if build.with? "malonzo"
+  if build.with? "ghc"
```

Comment:
- It's 2 lines (or 3 if you don't like the default message) instead of 6.
- Using `:optional` / `:recommended` is preferred and more idiomatic if the option name is also the name of the underlying formula. It used to be `without-malonzo` and `malonzo` isn't a formula. Since it's now `ghc`, the situation is different because `ghc` is a formula.